### PR TITLE
FIX: FormKit: Allow 0 in required number input

### PR DESCRIPTION
**Pull request by Lhcfl**
0 is falsy in JavaScript, so the original code would treat 0 as if it were not input. This unique exception was added to prevent 0 from being treated as empty input.

Screenshot before:

image

### CODE REVIEW

File: app/assets/javascripts/discourse/tests/integration/components/form-kit/field-test.gjs [hunk 1]
```diff
@@ -111,6 +111,10 @@ module("Integration | Component | FormKit | Field", function (hooks) {
     assert.form().hasErrors({ foo: ["Required"], bar: ["Required"] });
     assert.form().field("foo").hasError("Required");
     assert.form().field("bar").hasError("Required");
+
+    await fillIn("input[name=foo]", "0");
+    await formKit().submit();
+    assert.form().field("foo").hasNoError();
```
Comment (jjaffeux) [inline app/assets/javascripts/discourse/tests/integration/components/form-kit/field-test.gjs:117]: I would prefer we add a input-number-test.gjs file for this please

File: app/assets/javascripts/discourse/app/form-kit/lib/validator.js [hunk 2]
```diff
@@ -104,7 +104,7 @@ export default class Validator {
         }
         break;
       case "input-number":
-        if (!value || typeof value === "undefined" || isNaN(Number(value))) {
+        if ((!value && value !== 0) || isNaN(Number(value))) {
```
Comment (jjaffeux) [inline app/assets/javascripts/discourse/app/form-kit/lib/validator.js:107]: Ah  yes good catch thanks

File: app/assets/javascripts/discourse/tests/integration/components/form-kit/field-test.gjs (hunk 1 above)
Comment (Lhcfl) [reply app/assets/javascripts/discourse/tests/integration/components/form-kit/field-test.gjs:117]: how about put it into [input-test.gjs](https://github.com/discourse/discourse/blob/8d0b62c34446d96d4a98c7fa1c99a4098245fd2c/app/assets/javascripts/discourse/tests/integration/components/form-kit/controls/input-test.gjs)?

Comment (jjaffeux) [reply app/assets/javascripts/discourse/tests/integration/components/form-kit/field-test.gjs:117]: I would prefer input-number-test.gjs

Comment (jjaffeux) [reply app/assets/javascripts/discourse/tests/integration/components/form-kit/field-test.gjs:117]: And maybe I should just rename input-test.gjs to input-text-test.gjs

**Review by jjaffeux — APPROVED** [review]
Thanks for the fix and tests :rocket:

**Pull request merged**